stl/inc/algorithm

```diff
@@ -3010,6 +3010,65 @@ void replace(_ExPo&& _Exec, const _FwdIt _First, const _FwdIt _Last, const _Ty&
     const _Ty& _Newval) noexcept; // terminates
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+
+    // VARIABLE ranges::replace
+    class _Replace_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, sentinel_for<_It> _Se, class _Ty1, class _Ty2, class _Pj = identity>
+            requires indirectly_writable<_It, const _Ty2&> &&
+                     indirect_binary_predicate<equal_to, projected<_It, _Pj>, const _Ty1*>
+        constexpr _It operator()(
+            _It _First, _Se _Last, const _Ty1& _Oldval, const _Ty2& _Newval, _Pj _Proj = {}) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UResult = _Replace_unchecked(
+                _Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)), _Oldval, _Newval, _Pass_fn(_Proj));
+
+            _Seek_wrapped(_First, _STD move(_UResult));
+            return _STD move(_First);

```

Comment:
`_First` is implicitly movable since it's a variable with automatic storage duration ([class.copy.elision]/3), so the `move` here is extraneous (and on 3064). It isn't detrimental since we can't elide moves from a function parameter ([class.copy.elision]/1.1) but it's a good rule of thumb to never explicitly `move` when returning an automatic storage duration variable so we don't do so when it does matter (as it does on 3045, which _is_ a candidate for elision).

Reply:
Done